projects/ngx-datatable/src/lib/components/datatable.component.ts

```diff
@@ -366,13 +374,11 @@ export class DatatableComponent implements OnInit, DoCheck, AfterViewInit, After
   };
 
   /**
-   * Row specific classes.
-   * Similar implementation to ngClass.
-   *
-   *  [rowClass]="'first second'"
-   *  [rowClass]="{ 'first': true, 'second': true, 'third': false }"
+   * A function which is called with the row and should return either:
+   * - a string: `"class-1 class-2`
+   * - a Record<string, boolean>: `{ 'class-1': true, 'class-2': false }`
    */
-  @Input() rowClass: any;
+  @Input() rowClass: (row: Group<TRow> | TRow) => string | Record<string, boolean>;
```

Comment:
I think we shouldn't just limit it to be a function but also allow users to pass `string | Record<string, boolean>` directly.

Reply:
you are right in general. But the implementation currently only supports this function. We can do this in another MR.